### ISSUE TITLE
multiarch support

### ISSUE THREAD

**Issue by povilasv** [conprof/conprof#44]
I would be keen to add multi arch support for arm/arm64. We might as well do ppc64le / s390x . 

For quay you would need to create empty repos like I did for systemd_exporter:

https://quay.io/repository/povilasv/systemd-exporter-linux-amd64
https://quay.io/repository/povilasv/systemd-exporter-linux-s390x
https://quay.io/repository/povilasv/systemd-exporter-linux-ppc64le
https://quay.io/repository/povilasv/systemd-exporter-linux-armv7
https://quay.io/repository/povilasv/systemd-exporter-linux-ppc64le

This is how it looks in quay:

https://quay.io/repository/povilasv/systemd-exporter?tab=tags

![image](https://user-images.githubusercontent.com/22289110/79850048-05143980-83cc-11ea-9405-591145a3489e.png)

What do you think?

FYI I have a working k8s cluster which has arm64/arm nodes in it for testing.

**Comment by brancz** [comment]
Nice :+1: that would be really awesome!

**Comment by povilasv** [comment]
FYI cloud drone ci doesn't support docker in docker qemu https://discourse.drone.io/t/run-docker-container-in-docker/5704

So we will have to use workers that are native to that platform (no cross compiling). So we only get arm / arm64. One thing to note is that build queues for arm / arm64 are quite long.

Right now I have proof of concept working: 
https://cloud.drone.io/povilasv/conprof/12/1/1

I really like drone ci & it looks really good. But If we conisder switching to circle ci we can use https://github.com/prometheus/circleci prometheus orb.

**Comment by povilasv** [comment]
FYI build wait times got way lower, maybe it was just super busy, so we can ignore the wait time issue.

**Comment by povilasv** [comment]
![image](https://user-images.githubusercontent.com/22289110/80188100-60873700-8619-11ea-9687-d1e25dfe9be0.png)

Finally figured out a way to do it, will test on my cluster and do a PR.

turns out we don't need empty repos, as we just use tags for different architecture images.

**Comment by brancz** [comment]
Nice :+1:. Yeah if we can I would prefer to stick to drone, but no strong opinion. Looking forward to the contribution!

**Comment by povilasv** [comment]
Arm64 works beautifully

![image](https://user-images.githubusercontent.com/22289110/80273361-174aec00-86da-11ea-9c3e-547946d0ced4.png)

![image](https://user-images.githubusercontent.com/22289110/80273378-30ec3380-86da-11ea-8e86-53f2f2cf2b55.png)

**Comment by povilasv** [comment]
I've made it work on arm (32bit) with the following patch https://github.com/conprof/conprof/pull/45/files#diff-604f4192deea5adf9c159e5f88f64630R54-R69 

But CPU usage is crazy high:

![image](https://user-images.githubusercontent.com/22289110/80446210-ed363b80-891e-11ea-87c5-b104f1f5cb75.png)

compared to arm64:

![image](https://user-images.githubusercontent.com/22289110/80446364-5322c300-891f-11ea-9446-6ed47655e6c8.png)

Not sure what happened there, but I will run arm version for a bit longer and try to reproduce / look at profiles.

**Comment by povilasv** [comment]
Couldn't reproduce the cpu spike, so I propose to adding it in https://github.com/conprof/conprof/pull/49 and if it happens again I will investigate ;)

**Comment by povilasv** [comment]
Actually I got it again.

When I straced it forever looped in:
```
sched_yield()                           = 0
sched_yield()                           = 0
sched_yield()                           = 0
sched_yield()                           = 0
sched_yield()                           = 0
sched_yield()                           = 0
sched_yield()                           = 0
sched_yield()                           = 0
```

which is go runtime bug in https://github.com/golang/go/issues/38023 , which will be fixed in 1.14.2
